rclpy/test/test_parameter_client.py

```diff
@@ -0,0 +1,136 @@
+        ])
+        self.executor.spin_until_future_complete(future)
+        results = future.result()
+        assert results is not None
+        res = [i.successful for i in results.results]
+        assert all(res)
+
+    def test_get_parameter(self):
+        future = self.client.get_parameters(['int_arr_param', 'float/param//'])
+        self.executor.spin_until_future_complete(future)
+        results = future.result()
+        assert results is not None
+        assert list(results.values[0].integer_array_value) == [1, 2, 3]
+        assert results.values[1].double_value == 3.14
+
+    def test_list_parameters(self):
+        future = self.client.list_parameters()
+        self.executor.spin_until_future_complete(future)
+        results = future.result()
+        assert results is not None
+        assert 'int_arr_param' in results.result.names
+        assert 'float/param//' in results.result.names
+
+    def test_describe_parameters(self):
+        future = self.client.describe_parameters(['int_arr_param'])
+        self.executor.spin_until_future_complete(future)
+        results = future.result()
+        assert results is not None
+        assert results.descriptors[0].type == 7
+        assert results.descriptors[0].name == 'int_arr_param'
+
+    def test_get_paramter_types(self):
+        future = self.client.get_parameter_types(['int_arr_param'])
+        self.executor.spin_until_future_complete(future)
+        results = future.result()
+        assert results is not None
+        assert results.types[0] == 7  # refer to rcl_interfaces.msg.ParameterType
+
+    def test_set_parameters_atomically(self):
+        future = self.client.set_parameters_atomically([
+            Parameter('int_param', Parameter.Type.INTEGER, 888).to_parameter_msg(),
+            Parameter('string/param', Parameter.Type.STRING, 'Hello World').to_parameter_msg(),
+        ])
+        self.executor.spin_until_future_complete(future)
+        results = future.result()
+        assert results is not None
+        assert results.result.successful
+
+    def test_delete_parameters(self):
+        self.target_node.declare_parameter('delete_param', 10)
+        descriptor = rcl_interfaces.msg.ParameterDescriptor(dynamic_typing=True)
+        self.target_node.declare_parameter('delete_param_dynamic', 10, descriptor=descriptor)
+
+        future = self.client.delete_parameters(['delete_param'])
+        self.executor.spin_until_future_complete(future)
+        result = future.result()
+        assert result is not None
+        assert not result.results[0].successful
+        assert result.results[0].reason == 'Static parameter cannot be undeclared'
+
+        future = self.client.delete_parameters(['delete_param_dynamic'])
+        self.executor.spin_until_future_complete(future)
+        result = future.result()
+        assert result is not None
+        assert result.results[0].successful
```

Comment:
We could just form the assertions with a direct comparsion to the expected result. Quick fix if we want it but I didn't go this route because it was unnecessairly verbose.

```python
        assert results == rcl_interfaces.srv.DescribeParameters.Response(
            descriptors=[rcl_interfaces.msg.ParameterDescriptor(name='int_arr_param',
                                                                type=7,
                                                                description='',
                                                                additional_constraints='',
                                                                read_only=False,
                                                                dynamic_typing=False,
                                                                floating_point_range=[],
                                                                integer_range=[])])
```